resources/aws/security_group.go

```diff
@@ -57,67 +46,70 @@ func (s SecurityGroup) findExisting() (*ec2.SecurityGroup, error) {
 				},
 			},
 		},
+	})
+	if err != nil {
+		return nil, microerror.MaskAny(err)
 	}
 
-	for _, filter := range filters {
-		securityGroups, err := s.Clients.EC2.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{
-			Filters: filter,
-		})
-		if err != nil {
-			return nil, microerror.MaskAny(err)
-		}
-
-		if len(securityGroups.SecurityGroups) < 1 {
-			continue
-		}
-
-		return securityGroups.SecurityGroups[0], nil
-	}
-
-	return nil, microerror.MaskAny(securityGroupFindError)
-}
-
-func (s *SecurityGroup) checkIfExists() (bool, error) {
-	_, err := s.findExisting()
-	if IsSecurityGroupFind(err) {
-		return false, nil
-	} else if err != nil {
-		return false, microerror.MaskAny(err)
+	if len(securityGroups.SecurityGroups) < 1 {
+		return nil, microerror.MaskAny(securityGroupFindError)
 	}
 
-	return true, nil
+	return securityGroups.SecurityGroups[0], nil
 }
 
 func (s *SecurityGroup) CreateIfNotExists() (bool, error) {
-	exists, err := s.checkIfExists()
-	if err != nil {
-		return false, microerror.MaskAny(err)
-	}
-
-	if exists {
-		return false, nil
-	}
-
 	if err := s.CreateOrFail(); err != nil {
+		if strings.Contains(err.Error(), awsclient.SecurityGroupDuplicate) {
```

Comment:
Yes, there is no other way than matching the string in this case, as far as I know.